gdx/src/com/badlogic/gdx/math/Polygon.java

```diff
@@ -119,6 +119,17 @@ public void setVertices (float[] vertices) {
 		dirty = true;
 	}
 
+	 /** Set vertex position
+	  *
+	  * @param vertexNum min=0, max=(vertices/2)-1
+	  * @throws IllegalArgumentException if vertex doesnt exist */
+	 public void setVertex (int vertexNum, float x, float y) {
+		  if (vertexNum > localVertices.length / 2 -1) throw new IllegalArgumentException("the vertex " + vertexNum + "doesn't exist");

```

Comment:
I think the exception should be thrown if `vertexNum < 0` as well. And maybe an `IndexOutOfBoundsException` would fit better?

Reply:
I'd just add the `vertexNum < 0` check, the rest seems good to me.

Reply:
I add <0 check, i let you choose about exception to throw.

I got a question about to throw exception, 
Its really better to throw exception manually  than let an outofbound appears?
Or is just for stackstrace lisibility ?